File: 4기-코딩테스트-220608/4기 코딩테스트_220608/4기 - 1st 코딩테스트/Factorial/Factorial_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.Factorial.Factorial import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        n = 5

        # when
        result = solution(n)

        # then
        self.assertEqual(120, result)

    def test_2(self):
        # given
        n = 1

        # when
        result = solution(n)

        # then
        self.assertEqual(1, result)

    def test_3(self):
        # given
        n = 2

        # when
        result = solution(n)

        # then
        self.assertEqual(2, result)

    def test_4(self):
        # given
        n = 7

        # when
        result = solution(n)

        # then
        self.assertEqual(5040, result)

    def test_5(self):
        # given
        n = 10

        # when
        result = solution(n)

        # then
        self.assertEqual(3628800, result)

    def test_6(self):
        # given
        n = 12

        # when
        result = solution(n)

        # then
        self.assertEqual(479001600, result)


if __name__ == '__main__':
    unittest.main()

File: 4기 코딩테스트_220615/4기 - 1st 코딩테스트/Factorial/Factorial_test.js
import assert from 'assert'
import solution from './Factorial.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const n = 5

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(120, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const n = 1

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const n = 2

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(2, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const n = 7

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(5040, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const n = 10

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(3628800, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const n = 12

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(479001600, result)
    done();
  });
})
